src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml

```diff
@@ -64,21 +64,22 @@
   tags:
     - new_ansible_sudo_pass
 
-- name: Authorize current devs to login as ansible
-  become: yes
-  authorized_key: user=ansible state=present key="{{ lookup('file', authorized_keys_dir ~ item ~ '.pub') }}"
+- name: remove key using username
+  lineinfile:
+    dest: /home/ansible/.ssh/authorized_keys
+    state: absent
+    regexp: "username={{ item }}"
+  when: reset_user | default(False)
```

Comment:
If this option is not specified (e.g., because it is forgotten or the person running the command does not know to use it), then the retired key will not be removed. This is not ideal since we are using `update-user-key` more frequently than `update-users` these days because of performance, and therefore we are more likely to accidentally retain retired keys for a long time. Would it break anything if we change it to `default(True)`? I assume it will cause more churn and impact performance?

Reply:
Yes Daniel, using the default(True) will remove all user's the keys present in `{{ dev_users.present }}` first from the authorized_key file which are matching the with regex. 
I totally agree with you that the user might either forget or not might not be aware of using that option.